tools/utils/policy_test

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+import os
+import argparse
+import json
+import yaml
+import tests.common
+import c7n.credentials
+import c7n.utils
+import placebo
+import shutil
+import boto3
+
+parser = argparse.ArgumentParser('Policy Tester')
+parser.add_argument('--assume', dest='assume_role', help='Role to assume')
+subparsers = parser.add_subparsers(dest='command', help='Execution mode')
+subparsers.required=True
+
+record_parser = subparsers.add_parser('record')
+record_parser.add_argument('--cache', type=str, default='cache', help='Cache dir to write to')
+record_parser.add_argument('policy', type=argparse.FileType('r'), help='Policy file to consume')
+record_parser.add_argument('name', type=str, help='Name of the policy')
+
+test_parser = subparsers.add_parser('replay')
+test_parser.add_argument('--cache', type=str, default='cache' help='Cache dir to consume')
+test_parser.add_argument('policy', type=argparse.FileType('r'), help='Policy file to consume')
+test_parser.add_argument('name', type=str, help='Name of the policy')
+
+args = parser.parse_args()
+
+class TestPolicy(tests.zpill.PillTest):
```

Comment:
so we're not current distributing tests as part of the package, for cross provider usage we've tried to keep some common testing functionality in core (c7n/testing.py) but the recording mechanism is different for each provider atm.

Reply:
Can c7n just use `vcr` for everything? Looks like [this](https://github.com/kevin1024/vcrpy/issues/382) issue was fixed recently.

Reply:
our azure provider is pinned to an older version atm :-( so conflicting dep versions

Reply:
also gcp is using a bespoke http client (httplib2), though gcp also maintains shims to switch it to url lib3 (monkey of a monkey ~  😿

Reply:
Can the azure provider be upgraded?